ethers-providers/src/transports/ws.rs

```diff
@@ -401,8 +401,9 @@ where
             // Handle ws messages
             resp = self.ws.next() => match resp {
                 Some(Ok(resp)) => self.handle(resp).await?,
-                // TODO: Log the error?
-                Some(Err(_)) => {},
+                Some(Err(_)) => {
+                    return Err(ClientError::UnexpectedClose);
```

Comment:
maybe instead of dropping the error we can add a new field to enum `UnknownError(Err)` that wraps the error?

Reply:
That's a good idea. I'll try to get to that tomorrow.